web/src/hooks/useSearch.ts

```diff
@@ -55,121 +61,132 @@ export const extractAnswer = async (
   setCurrent: (e: CurrentSearch) => void
 ): Promise<SearchResult> => {
   var result: AssistantEntry = {
-    role: "assistant",
-    content: "",
+    role: 'assistant',
+    content: '',
     citations: [],
     citationsMap: new Map(),
-  };
-  var followups: Followup[] = [];
+  }
+  var followups: Followup[] = []
   for await (var data of iterateData(res)) {
     switch (data.state) {
-      case "loading":
-        setCurrent({ phase: data.phase, ...result });
-        break;
+      case 'loading':
+        setCurrent({phase: data.phase, ...result})
+        break
 
-      case "citations":
+      case 'citations':
         result = {
           ...result,
           citations: data?.citations || result?.citations || [],
-        };
-        setCurrent({ phase: data.phase, ...result });
-        break;
+        }
+        setCurrent({phase: data.phase, ...result})
+        break
 
-      case "streaming":
+      case 'streaming':
         // incrementally build up the response
-        const content = formatCitations((result?.content || "") + data.content);
+        const content = formatCitations((result?.content || '') + data.content)
         result = {
           content,
-          role: "assistant",
+          role: 'assistant',
           citations: result?.citations || [],
           citationsMap: findCitations(content, result?.citations || []),
-        };
-        setCurrent({ phase: "streaming", ...result });
-        break;
+        }
+        setCurrent({phase: 'streaming', ...result})
+        break
 
-      case "followups":
+      case 'followups':
         // add any potential followup questions
-        followups = data.followups.map((value: any) => value as Followup);
-        break;
-      case "done":
-        break;
-      case "error":
-        throw data.error;
+        followups = data.followups.map((value: any) => value as Followup)
+        break
+      case 'done':
+        break
+      case 'error':
+        throw data.error
     }
   }
-  return { result, followups };
-};
+  return {result, followups}
+}
 
 const fetchLLM = async (
   sessionId: string,
   query: string,
   settings: LLMSettings,
-  history: HistoryEntry[]
-): Promise<Response> =>
-  fetch(API_URL + "/chat", {
-    method: "POST",
-    cache: "no-cache",
+  history: HistoryEntry[],
+  controller: AbortController
+): Promise<Response | void> =>
+  fetch(API_URL + '/chat', {
+    signal: controller.signal,
```

Comment:
this will cause the `fetch` to be aborted if `controller.abort()` is called. Which can be done via the `searchbox` button. A nice thing about this is that it also works while iterating over the response data, so all that was really needed in this file was to add this line here (and for the stampy request) and add some error handling